### PR TITLE
Add Select component

### LINKED ISSUE #11: Create Select Component
- Use Cloud's `<FilterSelect />` as prior art
- Updated styles to match design mocks

### PR DESCRIPTION
**Pull request by jgiuffrida**
- Adds Select component, documentation, and tests (ontop of react-select)
- Updates index generation script to handle sub-directories

Resolves #11 

Preview: https://deploy-preview-20--laughing-shockley-2b3f4c.netlify.app/?path=/docs/components-forms-select--basic

### CODE REVIEW

File: components/Forms/Select/Select.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  options,
+  descriptionOptions,
+  groupedOptions,
+} from './select-story-data';
+
+<Meta title="Components/Forms/Select" component={Select} />
+
+# Select
+
+A single item select input built ontop of [React Select](https://react-select.com/home). Allows for searching, grouping (with labels), and descriptions on options.
```
Comment (aaronraff) [inline components/Forms/Select/Select.stories.mdx:23]:
```suggestion
A single item select input built on top of [React Select](https://react-select.com/home). Allows for searching, grouping (with labels), and descriptions on options.
```

Comment (aaronraff) [inline components/Forms/Select/Select.stories.mdx:23]:
```suggestion
A single item select input built on top of [React Select](https://react-select.com/home). Allows for searching, grouping (with labels), and descriptions on options.
```

File: components/Forms/Select/Select.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+### SelectOption
+
+Options are passed into `Select` as an array of `SelectOption` which is defined as:
+
+<Source
+  code={`interface SelectOption {
+  label: string;
+  value: string;
+  meta?: string;
+  options?: SelectOption[];
```
Comment (aaronraff) [inline components/Forms/Select/Select.stories.mdx:34]: Is this intended to be on the `SelectOption` interface?

**Review by aaronraff — COMMENTED** [review]
This makes sense to me! I just left one question for you 😄

File: components/Forms/Select/Select.stories.mdx (hunk 2 above)
Comment (jgiuffrida) [reply components/Forms/Select/Select.stories.mdx:34]: Yup - that's how grouped options work in react-select!

Comment (aaronraff) [reply components/Forms/Select/Select.stories.mdx:34]: Ah, okay! I'm seeing that in the example now, thanks!

**Review by aaronraff — APPROVED** [review]

**Review by amy-byrum — APPROVED** [review]
🥳

**Pull request merged**